src/Xamarin.Android.Build.Tasks/Utilities/NdkTools/WithClangNoBinutils.cs

```diff
@@ -14,6 +14,41 @@ public NdkToolsWithClangNoBinutils (string androidNdkPath, NdkVersion version, T
 		{
 			NdkToolNames[NdkToolKind.Linker] = "ld";
 			NoBinutils = true;
+
+			NeedClangWorkaround = IsWindows;
+			if (NeedClangWorkaround) {
+				//
+				// NDK r23 bug:
+				//
+				// The llvm toolchain directory contains a selection of shell scripts (both Unix and Windows)
+				// which call `clang/clang++` with different `-target` parameters depending on both the target
+				// architecture and API level. For instance, the clang/clang++ compilers targetting aarch64 on API level
+				// 28 will have the following Unix shell scripts present in the toolchain `bin` directory:
+				//
+				//   aarch64-linux-android28-clang.cmd
+				//   aarch64-linux-android28-clang++.cmd
+				//
+				// However, the Windows version of the NDK has a bug where there these scripts don't properly deal with
+				// spaces, which breaks the `BuildAotApplicationAndBundleAndÜmläüts()` unit tests, as it attempts to
+				// place the NDK into a directory containing spaces.  The result is:
+				//
+				//   CC="C:\a\_work\1\a\TestRelease\06-13_18.07.09\temp\SDK Ümläüts\ndk\toolchains\llvm\prebuilt\windows-x86_64\bin\armv7a-linux-androideabi19-clang.CMD"
+				//   AS="C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\Android\binutils\bin\arm-linux-androideabi-as.CMD"
+				//   …
+				//   [CC] "C:\a\_work\1\a\TestRelease\06-13_18.07.09\temp\SDK Ümläüts\ndk\toolchains\llvm\prebuilt\windows-x86_64\bin\armv7a-linux-androideabi19-clang.CMD" ^
+				//     -c -D__ANDROID_API__=19 -DANDROID -o obj\Release\bundles\armeabi-v7a\temp.o ^
+				//     -I "C:\a\_work\1\a\TestRelease\06-13_18.07.09\temp\SDK Ümläüts\ndk\toolchains\llvm\prebuilt\windows-x86_64\sysroot\usr\include\arm-linux-androideabi" ^
+				//     -I "C:\a\_work\1\a\TestRelease\06-13_18.07.09\temp\SDK Ümläüts\ndk\toolchains\llvm\prebuilt\windows-x86_64\sysroot\usr\include" ^
+				//     obj\Release\bundles\armeabi-v7a\temp.c
+				//   [cc stderr] 'C:\a\_work\1\a\TestRelease\06-13_18.07.09\temp\SDK' is not recognized as an internal or external command,
+				//   [cc stderr] operable program or batch file.
+				//
+				// The code below tries to rectify the situation by special-casing the compiler tool handling to
+				// return path to the actual .exe instead of the CMD.
+				//
+				NdkToolNames[NdkToolKind.CompilerC] = "clang.exe";
+				NdkToolNames[NdkToolKind.CompilerCPlusPlus] = "clang++.exe";
+			}
```

Comment:
This change won't be enough, it also needs [this code](https://github.com/xamarin/xamarin-android/blob/944f88a8b6f1c30eaf43f56cfbebf0f8cd7a547f/src/Xamarin.Android.Build.Tasks/Utilities/NdkTools/WithClangWithPlatforms.cs#L49-L68) to mimic the .cmd wrapper script